### PR TITLE
Add github templates.

### PR DESCRIPTION
**Pull request by Rapptz**
Opening this PR so people can look over wording and give some criticism and improvements. I'll keep this open for a while before finally merging it.

### CODE REVIEW

**Comment by laggycomputer** [comment]
I think `If there is a traceback, please show the entire thing` in the bug report should be changed to `Please show the traceback if any.` I don't see a reason to use `thing` over a more descriptive word or plain omission

**Comment by StarrFox** [comment]
on the pull request template's checklist wouldn't it make more sense for the first check to say "These changes have been tested if there was a code change." since the last check is if there was a code change or not

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+### Summary
+
+<!-- What is this pull request for? Does it fix any issues? -->
+
+### Checklist
+
+- [ ] If code changes were made then they have been tested.
+    - [ ] I have updated the documentation to reflect the changes.
+- [ ] This PR fixes an issue.
```
Comment (bryanforbes) [inline .github/PULL_REQUEST_TEMPLATE.md]: Some of these checkboxes could be solved with a "type" field:

**Type:** bug / feature / documentation

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
+### Summary
+
+<!-- What is this pull request for? Does it fix any issues? -->
+
+### Checklist
+
+- [ ] If code changes were made then they have been tested.
+    - [ ] I have updated the documentation to reflect the changes.
```
Comment (bryanforbes) [inline .github/PULL_REQUEST_TEMPLATE.md]: Should there be a checkbox about code style/linting?

Comment (Rapptz) [reply .github/PULL_REQUEST_TEMPLATE.md]: I think if this were to be added then I would need to also add a CONTRIBUTING.md file.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (Rapptz) [reply .github/PULL_REQUEST_TEMPLATE.md]: Not sure how I feel about this one.

Comment (bryanforbes) [reply .github/PULL_REQUEST_TEMPLATE.md]: That makes sense

**Pull request closed without merging**